Sources/ArcGISToolkit/FloorFilter/SiteFacilityPromptViewController.swift

```diff
@@ -0,0 +1,264 @@
+        siteFacilitySearchBar.tintColor = .customBlue
+    }
+    
+    public func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
+        isSearchActive = true
+        self.siteFacilitySearchBar.showsCancelButton = true
+    }
+    
+    public func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        isSearchActive = false
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+    }
+    
+    public func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+        isSearchActive = false
+        self.siteFacilitySearchBar.text = ""
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+    }
+    
+    public func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
+        isSearchActive = false
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+    }
+    
+    private func resetSearchFilteredResults() {
+        filteredSearchSites = viewModel.sites
+        filteredSearchFacilities = viewModel.facilities
+    }
+    
+    private func dismissSearchBar() {
+        self.isSearchActive = false
+        self.siteFacilitySearchBar.text = ""
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+        self.siteFacilitySearchBar.showsCancelButton = false
+    }
+}
+
+/// Extension for the Sites and Facilities Table View.
+extension SiteFacilityPromptViewController: UITableViewDataSource, UITableViewDelegate {
+    private func initializeSiteFacilityTableView() {
+        siteFacilityTableView.delegate = self
+        siteFacilityTableView.dataSource = self
+        siteFacilityTableView.separatorStyle = .none
+        siteFacilityTableView.register(UINib(nibName: "FloorFilterSiteFacilityCell", bundle: .module), forCellReuseIdentifier: "FloorFilterSiteFacilityCell")
+    }
+    
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        updatePromptTitle()
+        return isShowingFacilities ? filteredFacilities().count : filteredSites().count
+    }
+    
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = siteFacilityTableView.dequeueReusableCell(withIdentifier: "FloorFilterSiteFacilityCell", for: indexPath) as! SiteFacilityTableViewCell
+        let sites = filteredSites()
+        let facilities = filteredFacilities()
+        
+        // If there are no sites in the map, then directly show the facilities list.
+        if (sites.isEmpty) {
+            isShowingFacilities = true
+        }
+            
+        cell.siteFacilityDotImg.isHidden = true
+        cell.siteFacilityNameLabel?.font = UIFont(name:"Avenir", size:16)
+            
+        if (isShowingFacilities) {
+            if (indexPath.row <= facilities.count-1) {
```

Comment:
Please remove this check as if it fails, there is something catastrophically wrong, so we shouldn't fail gracefully (so that the error is caught and fixed right away in development). The last line of this method should be `return cell`, not `return UITableViewCell()`.